hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/authority/DefaultCAServer.java

```diff
@@ -224,12 +224,16 @@ private KeyPair getCAKeys() throws IOException {
         break;
       case KERBEROS_TRUSTED:
       case TESTING_AUTOMATIC:
-        X509CertificateHolder xcert = approver.sign(config,
-            getCAKeys().getPrivate(),
-            getCACertificate(), java.sql.Date.valueOf(beginDate),
-            java.sql.Date.valueOf(endDate), csr, scmID, clusterID);
-        store.storeValidCertificate(xcert.getSerialNumber(),
-            CertificateCodec.getX509Certificate(xcert));
+        X509CertificateHolder xcert;
+        try {
+          xcert = signAndStoreCertificate(beginDate, endDate, csr);
+        } catch (SCMSecurityException e) {
+          // Certificate with conflicting serial id, retry again may resolve
```

Comment:
why we would have duplicate serial id? is that because the serial id generator based on time where multiple csr requests at the same time?

Reply:
faced in our docker testing, multiple Dns may send csr simultaneously, SCM throws error for few of them saying serial id already exist. This small change resolves it but we should refactor it for SCM HA.

Reply:
Make senses to me. Let's open a follow up JIRA to have better serial ID generator.